### PR TITLE
Check for FreeType functions required for avatars

### LINKED ISSUE #7454: Nextcloud 13 expects FreeType support to be enabled in PHP GD
Since #6876 the system running Nextcloud should have FreeType support enabled in PHP GD; otherwise [`imagettfbbox`](https://github.com/nextcloud/server/blob/66f523e13ff98aef8cdca9f449df7353dcf60da7/lib/private/Avatar.php#L252) and [`imagettftext`](https://github.com/nextcloud/server/blob/66f523e13ff98aef8cdca9f449df7353dcf60da7/lib/private/Avatar.php#L258) would not be available (as explained in [PHP Documentation](http://php.net/manual/en/function.imagettftext.php#refsect1-function.imagettftext-notes)) and the request to get the avatar would fail with an internal error.

This only affects auto-generated avatars, and does not prevent Nextcloud from running, so it should be notified to the admin in the settings page when the requirement is not met.

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #7454

### CODE REVIEW

File: settings/Controller/CheckSetupController.php [hunk 1]
```diff
@@ -409,6 +409,18 @@ protected function isOpcacheProperlySetup() {
 		return $isOpcacheProperlySetUp;
 	}
 
+	/**
+	 * Check if the required FreeType functions are present
+	 * @return bool
+	 */
+	protected function hasFreeTypeSupport() {
+		if (function_exists('imagettfbbox') && function_exists('imagettftext')) {
```
Comment (ChristophWurst) [inline settings/Controller/CheckSetupController.php]: I'd prefer the shorter version
```php
return function_exists('imagettfbbox') && function_exists('imagettftext');
```

Comment (rullzer) [reply settings/Controller/CheckSetupController.php]: fair enough

**Review by ChristophWurst — APPROVED** [review]
Looks good!

**Review by danxuliu — APPROVED** [review]
Tested on a system without FreeType support enabled and works as expected 👍 

I have noticed, though, that not having FreeType support affects other elements besides just the avatars, for example the user settings UI (if you create a user or modify a user setting the UI will not be updated, although if you refresh the page you can see that it was modified as expected).

Maybe the error message should be something like "This will result in broken avatars and user settings UI" instead ("will" instead of "can" and "user settings UI" included in the message too)?

**Comment by rullzer** [comment]
@danxuliu I don't have a preference. Feel free to change the wording :)

**Comment by jancborchardt** [comment]
@danxuliu slightly improved:

> This will result in broken profile pictures and settings interface.

(We call it »profile picture« in the settings, should not use »avatar« anywhere. Also, no technical abbreviation like »UI«.)

**Comment by danxuliu** [comment]
I have amended the commits with @jancborchardt suggestion, thanks!

**Pull request merged**